src/store/modules/rate/index.js

```diff
@@ -25,8 +33,28 @@ const actions = {
         .then((res) => {
           const rates = res.data.result
           commit('setRates', rates)
+          commit('loadRates')
+          resolve(res)
+        })
+        .catch((err) => {
+          reject(err)
+        })
+    })
+  },
+  getHistoryRates ({ commit }, { date, coin }) {
+    const url = getEnpointUrl('infoservice')
+    return new Promise((resolve, reject) => {
+      axios
+        .get(`${url}/getHistory?timestamp=${date}&coin=${coin}`)
+        .then((res) => {
+          let rates
+          if (res.data.result[0]) {
+            rates = res.data.result[0].tickers
+          } else {
+            rates = undefined
+          }
+          commit('setHistoryRates', rates)
```

Comment:
I think we should cache rates by coin and by timestamp.
And return history rates often without unnecessary requests.